engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetInputTexture.java

```diff
@@ -111,40 +104,16 @@ public StateChange getDefaultInstance() {
         return defaultInstance;
     }
 
-    /**
-     * Instances of this class bind a texture to a texture unit. The integer identifying
-     * the texture unit is then passed to a shader program using the material/parameter
-     * pair provided on construction. See the source of the execute() method for the
-     * nitty gritty details.
-     *
-     * WARNING: RenderPipelineTasks are not meant for direct instantiation and manipulation.
-     * Modules or other parts of the engine should take advantage of them through classes
-     * inheriting from StateChange.
-     */
-    private class SetInputTextureTask implements RenderPipelineTask {
-        private final int textureSlot;
-        private final int textureId;
-        private final Material material;
-        private final String materialParameter;
-
-        private SetInputTextureTask(int textureSlot, int textureId, ResourceUrn materialURN, String materialParameter) {
-            this.textureSlot = textureSlot;
-            this.textureId = textureId;
-            this.material = getMaterial(materialURN);
-            this.materialParameter = materialParameter;
-        }
-
-        @Override
-        public void execute() {
-            glActiveTexture(GL_TEXTURE0 + textureSlot);
-            glBindTexture(GL_TEXTURE_2D, textureId);
-            material.setInt(materialParameter, textureSlot, true);
-        }
+    @Override
+    public String toString() {
+        return String.format("%30s: slot %s, texture %s, material %s, parameter %s", this.getClass().getSimpleName(),
+                textureSlot, textureId, material.getUrn().toString(), materialParameter);
+    }
 
-        @Override
-        public String toString() {
-            return String.format("%30s: slot %s, texture %s, material %s, parameter %s", this.getClass().getSimpleName(),
-                    textureSlot, textureId, material.getUrn().toString(), materialParameter);
-        }
+    @Override
+    public void process() {
```

Comment:
If you had listed the process() method before the toString() method github wouldn't be confused and wouldn't have flagged toString() as changed. This is something you should be able to notice also in the IDE. Again, one of the goal is to reduce the number of changed lines for the reviewer to go through as it all adds up. In this case I had to figure out what changed and it took me a minute or so to be sure nothing has.